src/NServiceBus.Core/Audit/Audit.cs

```diff
@@ -10,30 +12,34 @@ public class Audit : Feature
         internal Audit()
         {
             EnableByDefault();
-            Prerequisite(config =>
-                AuditConfigReader.GetConfiguredAuditQueue(config.Settings, out auditConfig),
-                "No configured audit queue was found");
+            Prerequisite(config =>AuditConfigReader.GetConfiguredAuditQueue(config.Settings, out auditConfig),"No configured audit queue was found");
         }
 
-        AuditConfigReader.Result auditConfig;
 
         /// <summary>
         /// See <see cref="Feature.Setup"/>
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.MainPipeline.Register<AuditBehavior.Registration>();
-            context.MainPipeline.Register<AttachCausationHeadersBehavior.Registration>();
+            context.Pipeline.Register<InvokeAuditPipelineBehavior.Registration>();
+            context.Pipeline.RegisterConnector<AuditToDispatchConnector>("Dispatches the audit message to the transport");
+         
+
+            context.Container.ConfigureComponent(b =>
```

Comment:
Agreed, me and @SzymonPobiega talked about a "pipeline factory". I'll schedule a call